components/firebase_admin_sdk/firebase_admin_sdk.app.js

```diff
@@ -0,0 +1,186 @@
+    },
+    _getHeaders(withAuth) {
+      const defaultHeader = {
+        "Content-Type": "applicaton/json",
+      };
+      const headers = withAuth
+        ? {
+          ...defaultHeader,
+          Authorization: `Bearer ${tokens.token}`,
+        }
+        : defaultHeader;
+      return headers;
+    },
+    async _makeRequest(method, url, data, params = {}, withAuth = true) {
+      const config = {
+        method,
+        url,
+        headers: this._getHeaders(withAuth),
+        data,
+        params,
+      };
+      return (await axios(config)).data;
+    },
+    /**
+     * Retrieves a Bearer token for use with the Firebase REST API.
+     * @param {string} apiKey - the Web API Key, which is obtained from the project
+     * settings page in the admin console
+     * @returns {object} returns an object containing a new token and refresh token
+     */
+    async _getToken(apiKey) {
+      const { clientEmail } = this.$auth;
+      const newCustomToken = await admin
+        .auth()
+        .createCustomToken(clientEmail)
+        .catch((error) => {
+          console.log("Error creating custom token:", error);
+        });
+      const data = {
+        token: newCustomToken,
+        returnSecureToken: true,
+      };
+      const params = {
+        key: apiKey,
+      };
+      return await this._makeRequest(
+        "POST",
+        "https://identitytoolkit.googleapis.com/v1/accounts:signInWithCustomToken",
+        data,
+        params,
+        false,
+      );
+    },
+    /**
+     * Exchanges a refresh Token for a new Bearer token for use with the Firebase REST API.
+     * @param {string} apiKey - the Web API Key, which is obtained from the project settings
+     * page in the admin console
+     * @returns {object} returns an object containing a new token and refresh token
+     */
+    async _refreshToken(apiKey) {
+      const data = {
+        grant_type: "refresh_token",
+        refresh_token: tokens.refreshToken,
+      };
+      const params = {
+        key: apiKey,
+      };
+      return await this._makeRequest(
+        "POST",
+        "https://securetoken.googleapis.com/v1/token",
+        data,
+        params,
+        false,
+      );
+    },
+    /**
+     * Gets a new brand new token or exchanges the refresh token for a new Bearer token
+     * for use with the Firebase REST API.
+     * @param {string} apiKey - the Web API Key, which is obtained from the project settings
+     * page in the admin console
+     */
+    async _getFreshTokens(apiKey) {
+      if (tokens) {
+        const {
+          id_token: it,
+          refresh_token: rt,
+        } = await this._refreshToken(apiKey);
+        tokens = {
+          token: it,
+          refreshToken: rt,
+        };
+        return;
+      }
+      const {
+        idToken,
+        refreshToken,
+      } = await this._getToken(
+        apiKey,
+      );
+      tokens = {
+        token: idToken,
+        refreshToken,
+      };
+    },
```

Comment:
This method might not be needed after all, unless there's a rate limit in terms of refreshing tokens too many times?

```suggestion
```